src/worker_files.rs

```diff
@@ -121,7 +121,7 @@ async fn async_pull_samples(
 ) {
     // We use async-await here, to better use IO stalls
     // We'll issue N async tasks in parallel, and wait for them to finish
-    let max_tasks_per_thread = min(num_cpus::get() * 2, limit);
+    let max_tasks = min(num_cpus::get() * 2, limit);
```

Comment:
naming was wrong, we have a threapool in the async runtime, and this value is the number of tasks we keep in flight